projects/hslayers/components/wfs-filter/wfs-filter.component.ts

```diff
@@ -0,0 +1,325 @@
+import {
+  getWfsAttributes,
+  getWfsUrl,
+  setWfsAttributes,
+} from 'hslayers-ng/common/extensions';
+import {takeUntilDestroyed, toSignal} from '@angular/core/rxjs-interop';
+
+@Component({
+  selector: 'hs-wfs-filter',
+  templateUrl: './wfs-filter.component.html',
+  styles: `
+    .hs-wfs-filter-panel > div {
+      padding: 0 0.75rem;
+    }
+  `,
+  standalone: true,
+  imports: [
+    NgClass,
+    HsFiltersComponent,
+    FormsModule,
+    AsyncPipe,
+    HsPanelHeaderComponent,
+    TranslateCustomPipe,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HsWfsFilterComponent extends HsPanelBaseComponent {
+  name = 'wfsFilter';
+
+  selectedLayer = signal<HsLayerDescriptor | null>(null);
+  rule = computed(() => this.selectedLayer()?.layer.get('wfsFilter') || {});
+
+  hsFiltersService = inject(HsFiltersService);
+  hsEventBusService = inject(HsEventBusService);
+  hsLayerManagerService = inject(HsLayerManagerService);
+  hsUtilsService = inject(HsUtilsService);
+  hsLayoutService = inject(HsLayoutService);
+  httpClient = inject(HttpClient);
+  hsLayerSelectorService = inject(HsLayerSelectorService);
+
+  availableLayers: Signal<HsLayerDescriptor[]>;
+
+  constructor() {
+    super();
+    const mainPanelStream = this.hsEventBusService.mapEventHandlersSet.pipe(
+      switchMap(() => this.hsLayoutService.mainpanel$),
+      filter((which) => which === 'wfsFilter'),
+      takeUntilDestroyed(),
+    );
+
+    this.availableLayers = toSignal(
+      mainPanelStream.pipe(
+        map(() => {
+          return this.hsLayerManagerService.data.layers.filter(
+            (l: HsLayerDescriptor) =>
+              this.hsUtilsService.instOf(l.layer, VectorLayer) &&
+              this.hsUtilsService.instOf(l.layer.getSource(), VectorSource) &&
+              getWfsUrl(l.layer),
+          );
+        }),
+        tap((layers) => {
+          const currentLayerTitle =
+            this.hsLayerSelectorService.currentLayer?.title;
+          if (currentLayerTitle) {
+            const layerToSelect = layers.find(
+              (layer) => layer.title === currentLayerTitle,
+            );
+            if (layerToSelect) {
+              this.selectLayer(layerToSelect);
+            }
+          }
+        }),
+      ),
+    );
+
+    // Use the same stream to update attributesExcludedFromList
+    mainPanelStream.subscribe(() => {
+      this.hsFiltersService.attributesExcludedFromList = [
+        'hs_normalized_IDW_value',
+        'boundedBy',
+      ];
+    });
+  }
+
+  /**
+   * Selects a layer and updates the filter service
+   * @param layer The layer to select
+   */
+  async selectLayer(layer: HsLayerDescriptor | null) {
+    this.selectedLayer.set(layer);
+    if (!layer) {
+      return;
+    }
+
+    const wfsUrl = getWfsUrl(layer.layer);
+    const url = this.hsUtilsService.proxify(
+      `${wfsUrl}?service=WFS&request=DescribeFeatureType&version=2.0.0&typeName=${layer.layer.get('layerName')}`,
```

Comment:
Why do you operate with `layerName` property? Existing property `name` is not sufficient?

Reply:
I am not exactly sure, will take a look at this once more. Might be just some leftover from me trying experimenting with something.

Reply:
There is bit of a mess with the layer name here
https://github.com/hslayers/hslayers-ng/blob/d9a306471c180f19130536cf1542e982889007da/projects/hslayers/services/add-data/url/wfs.service.ts#L514-L522

where we set both name and layerName (via options object) to the same value. I for some reason ended up using layerName. 
Theres also a note for thes property in type definition
https://github.com/hslayers/hslayers-ng/blob/d9a306471c180f19130536cf1542e982889007da/projects/hslayers/types/compositions/composition-layer-options.type.ts#L4
Anyways, name obtained via getName() should be perfectly fine and I guess preffered way to do so Ill make a change. Also it will make more sense to not double name related properties when defining wfs layer manually (in config for example)

Reply:
[80ccabf](https://github.com/hslayers/hslayers-ng/pull/5223/commits/80ccabf4be4efeb0f8fc23c75b5f9ca8353110d6)